### PR TITLE
fix: added enum validations to types

### PR DESCRIPTION
**Pull request by fran2810**
added enum validations to types:
 - secrets
 - users

### CODE REVIEW

File: src/types/user.go [hunk 1]
```diff
@@ -11,12 +11,12 @@ type UserIdentifier struct {
 }
 
 type UserInput struct {
-	Namespace string `json:"namespace" binding:"required"`
+	Namespace string `json:"namespace" binding:"required,oneof=admin viewer contributor"`
```
Comment (mzeevi) [inline src/types/user.go:14]: These are not the names of the roles we've created. `admin` already exists as a ClusterRole in Kubernetes. Should be `capp-admin`, `capp-viewer` and `capp-contributor`

File: src/types/user.go [hunk 2]
```diff
@@ -11,12 +11,12 @@ type UserIdentifier struct {
...
 	User
 }
 
 type PatchUserData struct {
-	Role string `json:"role" binding:"required" validate:"one0f=admin viewer contributor"`
+	Role string `json:"role" binding:"required,oneof=admin viewer contributor"`
```
Comment (mzeevi) [inline src/types/user.go:19]: Same here

**Review by mzeevi — CHANGES_REQUESTED** [review]

File: src/types/user.go (hunk 1 above)
Comment (fran2810) [reply src/types/user.go:14]: There's a map which converts this values in the controller I think for the platform it's much more clean and understands

**Review by mzeevi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**